### PR TITLE
docs(logging): review logging configuration

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
### All Submissions:

* [x] Have you followed the [documentation style guide](https://github.com/jina-ai/docs/tree/master/page_templates/style_guide.md)? 

### New Feature Submissions:

1. [x] Have you added your new file to a table of content index? 
2. [x] Did you use one of the templates in the [page_template](https://github.com/jina-ai/docs/tree/master/page_templates) folder?
3. [x] Is your title in Title Case? 
4. [x] Are your section headers in sentence case?

### CODE REVIEW

**Comment by Yongxuanzhang** [comment]
Preview link:
https://deploy-preview-135--jina-docs.netlify.app/chapters/logging.html

File: chapters/logging.rst [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        quiet: true
+      - uses: Pod2.yml
+
+
+What's more
+-------------
+
+FluentD
+^^^^^^^^^
+Fluentd is an open source data collector for unified logging layer [https://www.fluentd.org/](https://www.fluentd.org/).
```
Comment (nan-wang) [inline chapters/logging.rst]: I remember rst has a different way to write external hyperlinks. 

Reference:
 https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#internal-and-external-links

```suggestion
`Fluentd <https://github.com/fluent/fluentd>`_ is an open source data collector for unified logging layer.
```

File: chapters/logging.rst [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+Fluentd is expected to be used as a daemon receiving messages from the Jina logger and forwarding them to specific outputs using its
```
Comment (nan-wang) [inline chapters/logging.rst]:
```suggestion
`Fluentd <https://github.com/fluent/fluentd>`_ is expected to be used as a daemon receiving messages from the Jina logger and forwarding them to specific outputs using its
```

File: chapters/logging.rst [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    # this configuration describes where is the fluentD daemon running and waiting for logs to be emitted.
+    # FluentD then will have its own configuration to forward the messages according to its own syntax
+    # prefix will help fluentD filter data. This will be prepended for FluentD to easily filter incoming messages
+    tag: jina
+    profile-tag: jina-profile
+    host: 0.0.0.0
+    port: 24224
+
+
+To better understand fluentd configuration and to see how you can adapt to your needs, please see [https://docs.fluentd.org/configuration](https://docs.fluentd.org/configuration)
```
Comment (nan-wang) [inline chapters/logging.rst]:
```suggestion
To better understand fluentd configuration and to see how you can adapt to your needs, please see `Fluentd docs <https://docs.fluentd.org/configuration>`_
```

File: chapters/logging.rst [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+      append true
+      <buffer>
+          @type file
+          flush_mode interval
+          flush_interval 1s
+      </buffer>
+    </match>
+
+
+This is the default configuration, that works well together with the configuration provided in `logging.fluentd.yml`,
```
Comment (nan-wang) [inline chapters/logging.rst]:
```suggestion
This is the default configuration, that works well together with the configuration provided in ``logging.fluentd.yml``,
```

**Review by nan-wang — DISMISSED** [review]

File: chapters/logging.rst (hunk 1 above)
Comment (Yongxuanzhang) [reply chapters/logging.rst]: Yes. This section is written by markdown before. I just missed this link. Will update.

File: chapters/logging.rst (hunk 2 above)
Comment (Yongxuanzhang) [reply chapters/logging.rst]: Done

File: chapters/logging.rst (hunk 3 above)
Comment (Yongxuanzhang) [reply chapters/logging.rst]: Done

File: chapters/logging.rst (hunk 4 above)
Comment (Yongxuanzhang) [reply chapters/logging.rst]: Done

File: chapters/logging.rst [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+    :depth: 2
+
+Motivation
+-------------------
+
+In order to better understand, monitor, and debug the running and lifetime of Jina's peas, pods, and flow, Jina offers logging to help log the messages.
+
+Before you start
+-------------------
+You have installed the latest stable release of Jina Core according to the instructions found `here <https://docs.jina.ai/chapters/core/setup/index.html>`_ .
```
Comment (bwanglzu) [inline chapters/logging.rst]:
```suggestion
You have installed the latest stable release of Jina according to the instructions found `here <https://docs.jina.ai/chapters/core/setup/index.html>`_ .
```

File: chapters/logging.rst [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+
+Overview
+-------------------
+
+Jina logging message at first glance
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Let's take a look at a logging example first. This will help you quickly understand the content of Jina's logging message.
```
Comment (bwanglzu) [inline chapters/logging.rst]: I would rephrase `Let's take a look at a logging example first. This will help you quickly understand the content of Jina's logging message.` to `Let's start from a logging example`, how do you think?

File: chapters/logging.rst [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+
+Jina logs messages with different formats in different situations. This example follows one of the most used formats which you can view in the console.
```
Comment (bwanglzu) [inline chapters/logging.rst]:
```suggestion
Jina logs messages with different formats in different situations. The example below follows the most commonly used format:
```

File: chapters/logging.rst [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+    tag: jina
+    profile-tag: jina-profile
+    host: 0.0.0.0
+    port: 24224
+
+
+To better understand fluentd configuration and to see how you can adapt to your needs, please see `Fluentd docs <https://docs.fluentd.org/configuration>`_.
+
+Start fluentd daemon
+^^^^^^^^^^^^^^^
```
Comment (bwanglzu) [inline chapters/logging.rst]: this line a bit longer

**Review by bwanglzu — CHANGES_REQUESTED** [review]
I like it you convert it to reST, great work! Minor suggestions added.

File: chapters/logging.rst (hunk 6 above)
Comment (Yongxuanzhang) [reply chapters/logging.rst]: Sure! Will update

File: chapters/logging.rst (hunk 8 above)
Comment (Yongxuanzhang) [reply chapters/logging.rst]: Done

**Review by bwanglzu — APPROVED** [review]
lgtm! let's wait for @FionnD  's review then merge it

**Comment by Yongxuanzhang** [comment]
> lgtm! let's wait for @FionnD 's review then merge it

Thanks for your comments!

File: chapters/logging.rst [hunk 9]
```diff
@@ -0,0 +1,242 @@
+==============================
+Logging configuration in Jina
```
Comment (FionnD) [inline chapters/logging.rst]: **C**onfiguration 

We use title case for headers :)

File: chapters/logging.rst [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+   :description: A guide on logging configuration in Jina
+   :keywords: Jina, logging
+
+.. contents:: Table of Contents
+    :depth: 2
+
+Motivation
+-------------------
+
+In order to better understand, monitor, and debug the running and lifetime of Jina's peas, pods, and flow, Jina offers logging to help log the messages.
```
Comment (FionnD) [inline chapters/logging.rst]: In order to monitor and debug the 

The understanding is not needed :)

File: chapters/logging.rst [hunk 11]
```diff
@@ -0,0 +1,242 @@
...
+
+Before you start
+-------------------
+You have installed the latest stable release of Jina according to the instructions found `here <https://docs.jina.ai/chapters/core/setup/index.html>`_ .
+
+
+Overview
+-------------------
+
+Jina logging message at first glance
```
Comment (FionnD) [inline chapters/logging.rst]: messages

File: chapters/logging.rst [hunk 12]
```diff
@@ -0,0 +1,242 @@
...
+Let's start from an logging example. Jina logs messages with different formats in different situations. The example below follows the most commonly used format:
+
+.. highlight:: bash
+.. code-block:: bash
+
+   pod1@25908[I]:starting jina.peapods.runtimes.zmq.zed.ZEDRuntime...
+   pod1@25908[I]:input tcp://0.0.0.0:63915 (PULL_BIND) output tcp://0.0.0.0:63916 (PUSH_BIND) control over tcp://0.0.0.0:63914 (PAIR_BIND)
+   pod1@25903[S]:ready and listening
+
+It follows the following format:
```
Comment (FionnD) [inline chapters/logging.rst]: Replace with: 
This is how the message is composed :

File: chapters/logging.rst [hunk 13]
```diff
@@ -0,0 +1,242 @@
...
+
+.. highlight:: text
+.. code-block:: text
+
+    name@process[levelname]:message
+
+Where ``name`` is the name of the Pod, ``process`` is the process ID, ``levelname`` is the level.
+Jina logs messages in 6 different levels (DEBUG, INFO, WARNING, ERROR, CRITICAL, SUCCESS).
```
Comment (FionnD) [inline chapters/logging.rst]: Jina has six different levels of messages (DEBUG, INFO, WARNING, ERROR, CRITICAL, SUCCESS).

File: chapters/logging.rst [hunk 14]
```diff
@@ -0,0 +1,242 @@
...
+   * - FluentHandler
+     - Given the distributed nature of Jina's Peas and Pods, Jina offers a flexible solution that lets the user configure how and where the logs are forwarded.
+     - Need to set ``hostname``, ``process`` ``levelname``
+     - FluentD then will have its own configuration to forward the messages according to its own syntax
+
+
+Logging configuration in Action
+--------------------------------------
+
+Jina logging can be configured in YAML file.
```
Comment (FionnD) [inline chapters/logging.rst]: a YAML file.

File: chapters/logging.rst [hunk 15]
```diff
@@ -0,0 +1,242 @@
...
+.. code-block:: python
+
+        from jina.flow import Flow
+        from jina import Document
+
+        f = Flow(log_config='logging_cfg.yml').add().add()
+        # If you want to set the configuration for a certain Pod
+        #f = Flow().add(log_config='logging_cfg.yml').add()
+
+In YAML file you can customized the logger, choose the handler you need to monitor and save logs. ``logging_cfg.yml`` can be configured as follows:
```
Comment (FionnD) [inline chapters/logging.rst]: In a YAML file

File: chapters/logging.rst [hunk 16]
```diff
@@ -0,0 +1,242 @@
...
+FluentD
+^^^^^^^^^
+`Fluentd <https://github.com/fluent/fluentd>`_ is an open source data collector for unified logging layer.
+
+`Fluentd <https://github.com/fluent/fluentd>`_ is expected to be used as a daemon receiving messages from the Jina logger and forwarding them to specific outputs using its
+output plugins and configurations. 
+ 
+Although fluentd can be configured to forward logs to the user's preferred destinations, Jina offers a default configuration under `/resources` folder which expects a fluentd daemon to be running
+inside every machine running a Jina instance or Pea. Then the default configuration must be adapted to send the logs to the specific server
+where the Flow and the dashboard will be run. (This default behavior will evolve)
```
Comment (FionnD) [inline chapters/logging.rst]: Full stop at the end of line 165

File: chapters/logging.rst [hunk 17]
```diff
@@ -0,0 +1,242 @@
...
+
+See default `fluent.conf` configuration provided by Jina. It takes every input coming in the listening 24224 port and
```
Comment (FionnD) [inline chapters/logging.rst]: See **the** default

**Review by FionnD — CHANGES_REQUESTED** [review]
Great guide as well! Some minor grammar checks :)

File: chapters/logging.rst (hunk 9 above)
Comment (Yongxuanzhang) [reply chapters/logging.rst]: Sorry, I ignored this one. I thought it means to capitalize the first. Will keep that in mind. Thanks!

File: chapters/logging.rst (hunk 16 above)
Comment (Yongxuanzhang) [reply chapters/logging.rst]: Not sure about this one. You mean we should move the content of line 165 to 164?

File: chapters/logging.rst (hunk 17 above)
Comment (Yongxuanzhang) [reply chapters/logging.rst]: Done

File: chapters/logging.rst (hunk 15 above)
Comment (Yongxuanzhang) [reply chapters/logging.rst]: done

File: chapters/logging.rst (hunk 14 above)
Comment (Yongxuanzhang) [reply chapters/logging.rst]: done

File: chapters/logging.rst (hunk 13 above)
Comment (Yongxuanzhang) [reply chapters/logging.rst]: done

File: chapters/logging.rst (hunk 12 above)
Comment (Yongxuanzhang) [reply chapters/logging.rst]: done

File: chapters/logging.rst (hunk 11 above)
Comment (Yongxuanzhang) [reply chapters/logging.rst]: done

File: chapters/logging.rst (hunk 10 above)
Comment (Yongxuanzhang) [reply chapters/logging.rst]: done

**Review by FionnD — APPROVED** [review]
